setup.py

```diff
@@ -15,6 +15,7 @@
     "pathlib",
     "cloudpickle",
     "boto3",
+    "typing",
```

Comment:
Its the backport of the 3.5 functionality to earlier python versions 
https://pypi.python.org/pypi/typing/3.5.3.0 (lists 2.7). 

It can't provide the really nice type annotation syntax 3.5 allows, but at least the necessary types can be imported (e.g. `Option[...]`)